evm/vm/base.py

```diff
@@ -249,6 +249,11 @@ def get_transaction_class(cls):
     def validate_block(self, block):
         raise NotImplementedError("VM classes must implement this method")
 
+    @classmethod
+    @abstractmethod
+    def validate_header(cls, header: BlockHeader, parent_header: BlockHeader) -> None:
```

Comment:
Looking at how this is called in `validate_block`, shouldn't this be `parent_header: Optional[BlockHeader]`?